### PR TITLE
Attempt at refactoring to_currency into base

### PR DESCRIPTION
**Pull request by shulcsm**
### Status

- [x] READY
- [ ] HOLD
- [ ] WIP (Work-In-Progress)

### Additional notes
Attempt at generalizing and moving `to_currency` down as discussed in #133 
If this is fine i can do it to rest of mentioned languages.

However i'm not happy with `Num2Word_Base` and how it works.  I'm not sure how to do rest of integration.  What is the purpose of all these initialization callbacks? Shouldn't static things be better off as static properties?

### CODE REVIEW

File: num2words/lang_EN_EUR.py [hunk 1]
```diff
@@ -20,13 +20,7 @@
...
-                    **kwargs):
-        if 'currency' in kwargs:
-            return self._to_currency(val, **kwargs)
-
-        return self.to_splitnum(val, hightxt="euro/s", lowtxt="cents",
-                                jointxt=jointxt, longval=longval, cents=cents)
+    pass
 
 
 n2w = Num2Word_EN_EUR()
```
Comment (erozqba) [inline num2words/lang_EN_EUR.py]: We can now remove this file! :)

File: num2words/lang_EN_GB.py [hunk 2]
```diff
@@ -20,12 +20,7 @@
...
-    def to_currency(self, val, longval=True, **kwargs):
-        if 'currency' in kwargs:
-            return self._to_currency(val, **kwargs)
-
-        return self.to_splitnum(val, hightxt="pound/s", lowtxt="pence",
-                                jointxt="and", longval=longval)
+    pass
 
 
 n2w = Num2Word_EN_GB()
```
Comment (erozqba) [inline num2words/lang_EN_GB.py]: We can now remove this file! :)

**Comment by shulcsm** [comment]
I had some issues with moving uk doctests. Some of them don't pass, looks like something to do with one and feminine form. Someone native speaking should look into that.

**Comment by shulcsm** [comment]
I think i'm done here for now. Unless you have something to add. There is some duplication left in `to_cardinal` but i'm not sure what to do with it. I thin base needs some work, maybe after that.

File: num2words/lang_EN.py [hunk 3]
```diff
@@ -86,32 +88,9 @@ def to_year(self, val, longval=True):
...
-
-        return self.to_splitnum(val, hightxt="dollar/s", lowtxt="cent/s",
-                                jointxt="and", longval=longval, cents=True)
-
 
 n2w = Num2Word_EN()
 to_card = n2w.to_cardinal
 to_ord = n2w.to_ordinal
 to_ordnum = n2w.to_ordinal_num
 to_year = n2w.to_year
```
Comment (erozqba) [inline num2words/lang_EN.py]: We could remove this..

**Comment by shulcsm** [comment]
I don't want to make inheritance chain deeper and only way that would solve is get rid of duplication for `to_cardinal`. `_int2word` look similar but they are not. Same with `pluralize`.
If you really want to get rid of `to_cardinal` i can move it to mixin or something.
IMHO right solution for that would be pushing it down to base and generalizing all implementations but it's out of scope here.

**Comment by erozqba** [comment]
@shulcsm OK, you have convinced me, let's not make the inheritance chain deeper... We will try to push it to the base and generalize all implementations. I will only try to find the way or the person to verify the tests and implementation of UK before merge this and break something. I will try to look at this in the next day.
Thanks a lot again!

**Comment by shulcsm** [comment]
Ok. Failing tests have nothing to do with changes, though. It was in broken state before I touched it.

**Comment by erozqba** [comment]
@shulcsm if it is the case, then I will trust you here and accept this so we can move forward in fixing the test and pushing to_cadinal to the base class so we could remove the code duplication.

**Comment by shulcsm** [comment]
Thanks, here is an issue for tests: #138
You can also close some of the coverage issues, they have been solved with this.

**Comment by erozqba** [comment]
:+1:  thanks @shulcsm

**Pull request merged**